src/3ds_ui.cpp

```diff
@@ -204,17 +232,19 @@ void CtrUi::ProcessEvents() {
 		Input::Keys::N0, Input::Keys::N0, Input::Keys::PERIOD, Input::Keys::ADD
 	};
 
+	// reset state
+	touch_state =
+		touch_state == 1 ? 2 :
+		touch_state == 2 ? 2 : 0;
```

Comment:
This does not really make sense to me.

Reply:
This is for improving the frame rate by only redrawing the bottom screen when a touch happened (1) or one frame after a touch is finished (2). I can improve the naming.

Reply:
I understand what it should do, just the logic seems off here.
```bash
1 → 2
2 → 2 ?
0 → 0 ?
```

Reply:
When 2 it stays in 2 because the reset to 0 is in "UpdateDisplay". I can't reset it to 0 here because when there is a frameskip "UpdateDisplay" will miss the chance for a redraw.

Reply:
but maybe I can replace it with an if, because there is only one case where the value changes ;)

Reply:
replaced that nonsense with an if ^^', don't be asleep and code I guess ;)